DeprecationWarning for inspect.getargspec() raises Error/Fail in pytest-3.6

Running the tests for 2.0rc1 I am seeing 24 failures reported, all apparently based on the same
```
        warnings.warn("inspect.getargspec() is deprecated, "
                      "use inspect.signature() or inspect.getfullargspec()",
>                     DeprecationWarning, stacklevel=2)
E       DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()

../../../../../inspect.py:1041: DeprecationWarning
```
Setup:
```
platform darwin -- Python 3.6.1, pytest-3.0.5, py-1.4.32, pluggy-0.4.0

Running tests with Astropy version 2.0rc1.
Running tests in /sw/lib/python3.6/site-packages/astropy.

Date: 2017-06-29T01:55:41

Platform: Darwin-16.6.0-x86_64-i386-64bit

Executable: /sw/bin/python3.6

Full Python Version: 
3.6.1 (default, May  3 2017, 21:41:23) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.12.1
Scipy: 0.19.1
Matplotlib: 2.0.2
h5py: 2.7.0
Pandas: 0.20.2
Cython: 0.25.2
Using Astropy options: remote_data: none.

rootdir: /sw/lib/python3.6/site-packages/astropy, inifile: 
plugins: hypothesis-3.6.1, backports.unittest-mock-1.2.1
```
with `beautifulsoup 4.6.0`.

I could track it down to lxml actually, not beautifulsoup as I had first suspected:
```
________________________________________________________________ test_read_generic[t/html.html] ________________________________________________________________

filename = 't/html.html'

    @pytest.mark.parametrize('filename', files)
    def test_read_generic(filename):
>       Table.read(os.path.join(ROOT, filename), format='ascii')

/sw/lib/python3.6/site-packages/astropy/io/ascii/tests/test_connect.py:27: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/sw/lib/python3.6/site-packages/astropy/table/table.py:2521: in read
    out = io_registry.read(cls, *args, **kwargs)
/sw/lib/python3.6/site-packages/astropy/io/registry.py:531: in read
    data = reader(*args, **kwargs)
/sw/lib/python3.6/site-packages/astropy/io/ascii/connect.py:22: in read_asciitable
    return read(filename, **kwargs)
/sw/lib/python3.6/site-packages/astropy/io/ascii/ui.py:325: in read
    dat = _guess(table, new_kwargs, format, fast_reader_param)
/sw/lib/python3.6/site-packages/astropy/io/ascii/ui.py:473: in _guess
    dat = reader.read(table)
/sw/lib/python3.6/site-packages/astropy/io/ascii/html.py:345: in read
    return super(HTML, self).read(table)
/sw/lib/python3.6/site-packages/astropy/io/ascii/core.py:1159: in read
    self.lines = self.inputter.get_lines(table)
/sw/lib/python3.6/site-packages/astropy/io/ascii/core.py:310: in get_lines
    return self.process_lines(lines)
/sw/lib/python3.6/site-packages/astropy/io/ascii/html.py:96: in process_lines
    soup = BeautifulSoup('\n'.join(lines))
/sw/lib/python3.6/site-packages/bs4/__init__.py:228: in __init__
    self._feed()
/sw/lib/python3.6/site-packages/bs4/__init__.py:289: in _feed
    self.builder.feed(self.markup)
/sw/lib/python3.6/site-packages/bs4/builder/_lxml.py:250: in feed
    self.parser.feed(markup)
src/lxml/parser.pxi:1217: in lxml.etree._FeedParser.feed (src/lxml/lxml.etree.c:114563)
    ???
src/lxml/parser.pxi:1260: in lxml.etree._FeedParser.feed (src/lxml/lxml.etree.c:113522)
    ???
src/lxml/parser.pxi:850: in lxml.etree._BaseParser._getPushParserContext (src/lxml/lxml.etree.c:109144)
    ???
src/lxml/parser.pxi:867: in lxml.etree._BaseParser._createContext (src/lxml/lxml.etree.c:109352)
    ???
src/lxml/parsertarget.pxi:110: in lxml.etree._TargetParserContext._setTarget (src/lxml/lxml.etree.c:130710)
    ???
src/lxml/parsertarget.pxi:41: in lxml.etree._PythonSaxParserTarget.__cinit__ (src/lxml/lxml.etree.c:129176)
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

func = <bound method LXMLTreeBuilderForXML.start of <bs4.builder._lxml.LXMLTreeBuilder object at 0x11430ed30>>

    def getargspec(func):
        """Get the names and default values of a function's parameters.
    
        A tuple of four things is returned: (args, varargs, keywords, defaults).
        'args' is a list of the argument names, including keyword-only argument names.
        'varargs' and 'keywords' are the names of the * and ** parameters or None.
        'defaults' is an n-tuple of the default values of the last n parameters.
    
        This function is deprecated, as it does not support annotations or
        keyword-only parameters and will raise ValueError if either is present
        on the supplied callable.
    
        For a more structured introspection API, use inspect.signature() instead.
    
        Alternatively, use getfullargspec() for an API with a similar namedtuple
        based interface, but full support for annotations and keyword-only
        parameters.
        """
        warnings.warn("inspect.getargspec() is deprecated, "
                      "use inspect.signature() or inspect.getfullargspec()",
>                     DeprecationWarning, stacklevel=2)
E       DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()

/sw/lib/python3.6/inspect.py:1041: DeprecationWarning
```
The real culprit is this function in lxml-3.8.0's `parser target.pxi`:
```
cdef object inspect_getargspec
try:
    from inspect import getargspec as inspect_getargspec
except ImportError:
    from inspect import getfullargspec as inspect_getargspec
```
This might be their problem upstream then, as they should perhaps try `getfullargspec` (which exists at least as of 3.4) first, but I don't quite understand why the `DeprecationWarning` (which already is present in Python3.5's `inspect`) raises an error in the Python3.6 test suite - anything one might change in the test setup to avoid this?

Unfortunately it seems difficult to catch the warning in the test.
A bit confusingly to me, pytest's
```
E       DeprecationWarning: inspect.getargspec() is deprecated, use inspect.signature() or inspect.getfullargspec()
```
made it look as it is promoted to an `Error` somewhere along the way down, but that just seems to be the reporting format. Anyway the decorator approach just fails to raise the warning:
```
                with pytest.warns(DeprecationWarning):
>                   table = ascii.read(testfile['name'], guess=True, **guess_opts)
E                   Failed: DID NOT WARN

test_read.py:180: Failed
```

--- a/astropy/tests/helper.py
+++ b/astropy/tests/helper.py
@@ -146,11 +146,16 @@ _warnings_to_ignore_by_pyver = {
         r"The value of convert_charrefs will become True in 3\.5\. "
         r"You are encouraged to set the value explicitly\."]),
     (3, 5): set([
-        # py.test raises this warning on Python 3.5.
-        # This can be removed when fixed in py.test.
+        # py.test raised this warning in inspect on Python 3.5.
         # See https://github.com/pytest-dev/pytest/pull/1009
+        # Keeping it since e.g. lxml as of 3.8.0 is still calling getargspec()
         r"inspect\.getargspec\(\) is deprecated, use "
-        r"inspect\.signature\(\) instead"])}
+        r"inspect\.signature\(\) instead"]),
+    (3, 6): set([
+        # inspect raises this slightly different warning on Python 3.6.
+        # Keeping it since e.g. lxml as of 3.8.0 is still calling getargspec()
+        r"inspect\.getargspec\(\) is deprecated, use "
+        r"inspect\.signature\(\) or inspect\.getfullargspec\(\)"])}
 
 
 def enable_deprecations_as_exceptions(include_astropy_deprecations=True,

